### PR TITLE
feat: add prototype pollution exploit in typeorm

### PR DESCRIPTION
**Pull request by lirantal**

### CODE REVIEW

File: exploits/prototype-pollution-typeorm.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+Prototype Pollution in TypeORM can lead to SQL Injection attacks in both MongoDB and MySQL types of databases, as well as Denial of Service or escalate to other types of attacks for a vulnerable Node.js web application.
+
+## Requirements
+
+You'll need a MySQL server running. Try this:
+
+```sh
+# use mysql:5 as that still supports the good old native password mechanism to login
+# and keep things simple for the demo
+docker run -p3306:3306 --rm --name mysqld -e MYSQL_ROOT_PASSWORD=root mysql:5
```
Comment (Kirill89) [inline exploits/prototype-pollution-typeorm.md:12]: Please update `docker-compose.yml` as well.

File: exploits/prototype-pollution-typeorm.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+```
+
+For the MySQL client (to browse database records and such):
+```sh
+docker exec -it mysqld mysql -uroot -p
+```
+
+## Setup
+
+The root file `./typeorm-db.js` gets called when the app starts and seeds the database with 2 sample users.
```
Comment (Kirill89) [inline exploits/prototype-pollution-typeorm.md:22]:
```
failed connecting and seeding users to the MySQL database
Error: ER_BAD_DB_ERROR: Unknown database 'acme'
```

Looks like `create database acme;` step is missing.

File: typeorm-db.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+  const dbConnection = typeorm.getConnection('mysql')
+
+  const repo = dbConnection.getRepository("Users")
+  return repo
+}).then((repo) => {
+
+
+  console.log('Seeding 2 users to MySQL users table: Liran (role: user), Simon (role: admin')
+  const inserts = [
```
Comment (Kirill89) [inline typeorm-db.js:29]: It doesn't work with `() =>`. Please replace with

```
const inserts = [
      repo.insert({
        name: "Liran",
        address: "Israel",
        role: "user"
      }),
      repo.insert({
        name: "Simon",
        address: "UK",
        role: "admin"
      })
    ];
```

**Comment by lirantal** [comment]
@Kirill89 updated per your comments on https://github.com/snyk/goof/pull/784/commits/8719b249647552a1d85452fdb45dde546e28673d

**Pull request merged**